diff --git a/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_conn_cfg.hjson b/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_conn_cfg.hjson
index 52a54b9..5676c4f 100644
--- a/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_conn_cfg.hjson
+++ b/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_conn_cfg.hjson
@@ -1,9 +1,9 @@
-// Copyright lowRISC contributors (OpenTitan project).
+// Copyright lowRISC contributors.
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 {
   sub_flow:    conn
-  import_cfgs: ["{proj_root}/hw/formal/tools/dvsim/common_formal_cfg.hjson"]
+  import_cfgs: ["{proj_root}/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_formal_cfg.hjson"]
   dut:         "{name}"

   // Vars that need to exported to the env
@@ -18,4 +18,4 @@
       value: "default"
     }
   ]
-}
+}
\ No newline at end of file
diff --git a/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_formal_cfg.hjson b/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_formal_cfg.hjson
index afff2bb..bac1387 100644
--- a/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_formal_cfg.hjson
+++ b/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_formal_cfg.hjson
@@ -1,9 +1,9 @@
-// Copyright lowRISC contributors (OpenTitan project).
+// Copyright lowRISC contributors.
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 {
   flow:             formal
-  formal_root:      "{proj_root}/hw/formal"
+  formal_root:      "{proj_root}/hw/vendor/lowrisc_ip/formal"
   flow_makefile:    "{formal_root}/tools/dvsim/formal.mk"

   import_cfgs:      [// common server configuration for results upload
diff --git a/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_fpv_cfg.hjson b/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_fpv_cfg.hjson
index 4a92376..e8e8bef 100644
--- a/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_fpv_cfg.hjson
+++ b/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_fpv_cfg.hjson
@@ -1,9 +1,9 @@
-// Copyright lowRISC contributors (OpenTitan project).
+// Copyright lowRISC contributors.
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 {
   sub_flow: fpv
-  import_cfgs: ["{proj_root}/hw/formal/tools/dvsim/common_formal_cfg.hjson"]
+  import_cfgs: ["{proj_root}/hw/vendor/lowrisc_ip/formal/tools/dvsim/common_formal_cfg.hjson"]
   stopats: []
   task: ""
   after_load: []
@@ -14,4 +14,4 @@
     {TASK:       '{task}'},
     {AFTER_LOAD: '{after_load}'},
   ]
-}
+}
\ No newline at end of file