diff --git a/data/rom_ctrl_testplan.hjson b/data/rom_ctrl_testplan.hjson
index 71e2a2d..db8d78a 100644
--- a/data/rom_ctrl_testplan.hjson
+++ b/data/rom_ctrl_testplan.hjson
@@ -3,14 +3,14 @@
 // SPDX-License-Identifier: Apache-2.0
 {
   name: "rom_ctrl"
-  import_testplans: ["hw/dv/tools/dvsim/testplans/csr_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/alert_test_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/mem_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/passthru_mem_intg_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/tl_device_access_types_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/stress_all_with_reset_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson",
+  import_testplans: ["hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/csr_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/alert_test_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/mem_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/passthru_mem_intg_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/tl_device_access_types_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/sec_cm_fsm_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/stress_all_with_reset_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/sec_cm_count_testplan.hjson",
                      "rom_ctrl_sec_cm_testplan.hjson"]
   testpoints: [
     {
diff --git a/util/scramble_image.py b/util/scramble_image.py
index 91701d47..26f2f691 100755
--- a/util/scramble_image.py
+++ b/util/scramble_image.py
@@ -12,6 +12,15 @@ from typing import Dict, IO, Optional, Tuple
 import hjson  # type: ignore
 from Crypto.Hash import cSHAKE256

+# Add to the PATH the directory that this script and its "mem" dependency
+# expect to be able to use relative paths to import other dependencies from.
+from pathlib import Path
+PROJECT_ROOT = Path.resolve(Path(__file__)).parent.parent.parent.parent.parent.parent.parent
+LOWRISC_IP_ROOT = PROJECT_ROOT / Path("hw/vendor/lowrisc_ip")
+print("PROJECT_ROOT: ", PROJECT_ROOT)
+print("LOWRISC_IP_ROOT:", LOWRISC_IP_ROOT)
+sys.path.append(str(LOWRISC_IP_ROOT))
+
 from mem import MemChunk, MemFile
 from util.design.prince import prince, sbox  # type: ignore
 from util.design.secded_gen import ecc_encode_some  # type: ignore
