diff --git a/data/i2c_testplan.hjson b/data/i2c_testplan.hjson
index 1fc121e..a6a2c84 100644
--- a/data/i2c_testplan.hjson
+++ b/data/i2c_testplan.hjson
@@ -3,10 +3,10 @@
 // SPDX-License-Identifier: Apache-2.0
 {
   name: "i2c"
-  import_testplans: ["hw/dv/tools/dvsim/testplans/csr_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/alert_test_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/intr_test_testplan.hjson",
-                     "hw/dv/tools/dvsim/testplans/tl_device_access_types_testplan.hjson",
+  import_testplans: ["hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/csr_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/alert_test_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/intr_test_testplan.hjson",
+                     "hw/vendor/lowrisc_ip/dv/tools/dvsim/testplans/tl_device_access_types_testplan.hjson",
                      "i2c_sec_cm_testplan.hjson"]
   testpoints: [
     //-----------------------------------------------
