diff --git a/tb/tb.sv b/tb/tb.sv
--- a/tb/tb.sv
+++ b/tb/tb.sv
@@ -79,8 +79,8 @@
   // dut
   i2c dut (
     .clk_i                   (clk        ),
     .rst_ni                  (rst_n      ),
-    .ram_cfg_i               (prim_ram_1p_pkg::RAM_1P_CFG_REQ_DEFAULT),
-    .ram_cfg_o               (                                       ),
+    .ram_cfg_i               (prim_ram_1p_pkg::RAM_1P_CFG_DEFAULT),
+    .ram_cfg_rsp_o           (                                       ),

     .tl_i                    (tl_if.h2d  ),
