diff --git a/core/include/cv64a6_imafdchzcheri_sv39_config_pkg.sv b/core/include/cv64a6_imafdchzcheri_sv39_config_pkg.sv
index 683329b..dfe479a 100644
--- a/core/include/cv64a6_imafdchzcheri_sv39_config_pkg.sv
+++ b/core/include/cv64a6_imafdchzcheri_sv39_config_pkg.sv
@@ -128,7 +128,7 @@ package cva6_config_pkg;
       BPType: config_pkg::BHT,
       BHTEntries: unsigned'(CVA6ConfigBHTEntries),
       BHTHist: unsigned'(3),
-      DmBaseAddress: 64'h0,
+      DmBaseAddress: 64'h2000_0000,
       TvalEn: bit'(CVA6ConfigTvalEn),
       DirectVecOnly: bit'(0),
       NrPMPEntries: unsigned'(CVA6ConfigNrPMPEntries),
@@ -141,11 +141,11 @@ package cva6_config_pkg;
       NonIdempotentAddrBase: 1024'({64'b0, 64'b0}),
       NonIdempotentLength: 1024'({64'b0, 64'b0}),
       NrExecuteRegionRules: unsigned'(3),
-      ExecuteRegionAddrBase: 1024'({64'h8000_0000, 64'h10_0000, 64'h0}),
-      ExecuteRegionLength: 1024'({64'h40000000, 64'h10_0000, 64'h1000}),
+      ExecuteRegionAddrBase: 1024'({64'h8000_0000, 64'h1000_0000, 64'h2000_0000}),
+      ExecuteRegionLength: 1024'({64'h4000_0000, 64'h2_0000, 64'h1000}),
       NrCachedRegionRules: unsigned'(1),
-      CachedRegionAddrBase: 1024'({64'hE000_0000}),
-      CachedRegionLength: 1024'({64'h1000_0000}),
+      CachedRegionAddrBase: 1024'({64'h8000_0000}),
+      CachedRegionLength: 1024'({64'h4000_0000}),
       MaxOutstandingStores: unsigned'(7),
       DebugEn: bit'(1),
       AxiBurstWriteEn: bit'(1),
